cli/internal/publish/images/images.go

```diff
@@ -39,6 +41,25 @@ type listKey struct {
 	name, sum string
 }
 
+func QuickContentType(filename string) (string, error) {
+	filebytes := make([]byte, 512)
+
+	fp, err := os.Open(filename)
+	if err != nil {
+		return "", fmt.Errorf("failed to open file: %w", err)
+	}
+	defer fp.Close()
+	if _, err := fp.Read(filebytes); err != nil {
+		return "", fmt.Errorf("failed to read file: %w", err)
+	}
+
+	contentType := mime.TypeByExtension(filepath.Ext(filename))
+	if contentType == "" {
+		contentType = http.DetectContentType(filebytes)
```

Comment:
Maybe open and read file only `if contentType == ""`? Reverse the if and do an early return, otherwise read the first 512 and use the http detect method.

Reply:
Fixed in https://github.com/cloudquery/cloudquery/pull/19373/commits/269d066b6501c4059bb52924177756c35ad5bbb9